base_layer/core/src/transaction_builder.rs

```diff
@@ -0,0 +1,178 @@
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE
+
+use crate::{
+    block::AggregateBody,
+    transaction::{Transaction, TransactionError, TransactionInput, TransactionKernel, TransactionOutput},
+    types::{Base, BlindingFactor},
+};
+
+pub struct TransactionBuilder {
+    base: &'static Base,
+    body: AggregateBody,
+    offset: Option<BlindingFactor>,
+}
+
+impl TransactionBuilder {
+    /// Create an new empty TransactionBuilder
+    pub fn new(base: &'static Base) -> Self {
+        Self { base, offset: None, body: AggregateBody::empty() }
+    }
+
+    /// Update the offset of an existing transaction
+    pub fn with_offset(mut self, offset: BlindingFactor) -> Self {
+        self.offset = Some(offset);
+        self
+    }
+
+    /// Add an input to an existing transaction
+    pub fn with_input(mut self, input: TransactionInput) -> Self {
+        self.body = self.body.with_input(input);
+        self
+    }
+
+    /// Add an output to an existing transaction
+    pub fn with_output(mut self, output: TransactionOutput) -> Self {
+        self.body = self.body.with_output(output);
+        self
+    }
+
+    /// Add a series of inputs to an existing transaction
+    pub fn with_inputs(mut self, inputs: Vec<TransactionInput>) -> Self {
+        self.body = self.body.with_inputs(inputs);
+        self
+    }
+
+    /// Add a series of outputs to an existing transaction
+    pub fn with_outputs(mut self, outputs: Vec<TransactionOutput>) -> Self {
+        self.body = self.body.with_outputs(outputs);
+        self
+    }
+
+    /// Set the kernel of a transaction. Currently only one kernel is allowed per transaction
+    pub fn with_kernel(mut self, kernel: TransactionKernel) -> Self {
+        self.body = self.body.set_kernel(kernel);
+        self
+    }
+
+    pub fn finish(&self) -> Result<Transaction, TransactionError> {
```

Comment:
Nit: this method is usually called `build` in the builder pattern

Reply:
Sure, done.